docs/upgrade.html

```diff
@@ -139,7 +139,9 @@ <h5><a id="upgrade_420_notable" href="#upgrade_420_notable">Notable changes in 4
                 <ul>
                     <li>Null values are no longer accepted for most LIST-type configurations, except those that explicitly
                         allow a null default value or where a null value has a well-defined semantic meaning.</li>
-                    <li>Duplicate entries within the same list are no longer permitted.</li>
+                    <li>Most LIST-type configurations no longer accept duplicate entries, except in cases where duplicates
```

Comment:
> except those that explicitly allow a null default value or where a null value has a well-defined semantic meaning.

Null is only allowed when it's the default, right? There is no way to configure a null value explicitly in a config file.

> However, if users configure duplicate entries, the internal deduplication logic will still handle them.

If users configure duplicate entries when they are not accepted, they will be ignored and a warning will be logged.